### PR TITLE
Replacing the backend database

### LINKED ISSUE #57: Show total records
It would be very useful to show the total number of records when querying the data API.

### PR DESCRIPTION
**Pull request by dschowta**
Has following addition of features:
1. sqlite for backend
2. Code cleanups
3. Ability to query multiple series in parallel
4. Closes #57 

Open TODOs:
1. Units are not part of senml now. 
2. Validation of datasources during creation and updating
4. Unit test cases

### CODE REVIEW

**Comment by dschowta** [comment]
also closes #57

File: data/sqldb.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+		stmt := fmt.Sprintf("REPLACE INTO [%s] (time, value) VALUES %s",
+			ds, strings.Join(valueStrings, ","))
+		_, err := s.pool.Exec(stmt, valueArgs...)
+		return err
+	}
+
+	return nil
+}
+
+//This function converts a floating point number (which is supported by senml) to a bytearray
```
Comment (farshidtz) [inline data/sqldb.go]: is this returning "bytearray"?  Also check the comment on the next function.

File: data/sqldb.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	}
+	return int64ToFloatTime(t.UnixNano())
+}
+
+func FromSenmlTime(t float64) time.Time {
+	return time.Unix(0, floatTimeToInt64(t))
+}
+
+//caution: this function modifies the record and baseRecord values
+func denormalizeRecord(record *senml.Record, baseRecord **senml.Record, mask DenormMask) {
```
Comment (farshidtz) [inline data/sqldb.go]: why baseRecord is pointer to pointer?

File: data/storage.go [hunk 3]
```diff
@@ -22,6 +23,7 @@ func SupportedBackends(name string) bool {
 		INFLUXDB:   false, // Influxdb is not supported from HDS v1.0.0
 		MONGODB:    false, // Mongodb is not supported after HDS v0.5.3
 		SENMLSTORE: true,
```
Comment (farshidtz) [inline data/storage.go]: should be false

File: data/sqldb.go (hunk 2 above)
Comment (dschowta) [reply data/sqldb.go]: Base record stores the address of first record for future use. Storing just the value does not work unless a deep copy is performed.

File: data/sqldb.go (hunk 1 above)
Comment (dschowta) [reply data/sqldb.go]: was wrong. correcting it.

File: data/storage.go (hunk 3 above)
Comment (dschowta) [reply data/storage.go]: correcting it

**Pull request merged**